test/libyul/yulOptimizerTests/stackLimitEvader/too_many_params.yul

```diff
@@ -0,0 +1,32 @@
+{
+    function f(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13, a14, a15, a16, a17) {}
+    pop(memoryguard(0))
+    f(1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17)
+}
+// ----
+// step: stackLimitEvader
+//
+// {
+//     function f(a1, a2, a3, a4, a5, a6, a7, a9, a10, a11, a12, a13, a14, a15, a16, a17)
+//     { }
+//     pop(memoryguard(0x20))
+//     let _2 := 17
+//     let _3 := 16
+//     let _4 := 15
+//     let _5 := 14
+//     let _6 := 13
+//     let _7 := 12
+//     let _8 := 11
+//     let _9 := 10
+//     let _10 := 9
+//     let _11 := 8
+//     let _12 := 7
+//     let _13 := 6
+//     let _14 := 5
+//     let _15 := 4
+//     let _16 := 3
+//     let _17 := 2
+//     let _18 := 1
+//     mstore(0x00, _11)
```

Comment:
This prevents the ExpressionJoiner from moving the above back into argument position below...
And it doesn't look optimal... I'm not sure we can ensure that all arguments will remain reachable in the call below.
I'll need to think about this some more.

Reply:
Thinking about this some more... return parameters are handled fine by the PR already I think, but I may need to rethink parameters a bit, I'm not sure this will work out in general...

Reply:
Note: idea for test case for later: something around this: ``f(f(0,1,2), f(3,4,5), f(6,7,8))``.